### PR TITLE
Time Source was added to Cache.Builder

### PR DESCRIPTION
**Pull request by Komdosh**
HotFix for this Issue https://github.com/ReactiveCircus/cache4k/issues/19

Now `FakeTimeSource` in test directory. Users can provide any time sources.

### CODE REVIEW

**Review by ychescale9 — COMMENTED** [review]
Thanks! Could you also update the [snippet in README.md](https://github.com/ReactiveCircus/cache4k#unit-testing-cache-expirations)?

File: cache4k/src/commonMain/kotlin/io/github/reactivecircus/cache4k/Cache.kt [hunk 1]
```diff
@@ -79,12 +79,11 @@ public interface Cache<in Key : Any, Value : Any> {
         public fun maximumCacheSize(size: Long): Builder
 
         /**
-         * Specifies a [FakeTimeSource] for programmatically advancing the reading of the underlying
-         * [TimeSource] used for expiry checks in tests.
+         * Specifies a [TimeSource] provided externally.
          *
-         * If not specified, [TimeSource.Monotonic] will be used for expiry checks.
+         * If not specified, [TimeSource.Monotonic] will be used.
```
[the remark this answers is not in the document]
Comment (Komdosh) [reply cache4k/src/commonMain/kotlin/io/github/reactivecircus/cache4k/Cache.kt]: Yes I changed snippet

https://github.com/ReactiveCircus/cache4k/pull/20/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5 #

**Comment by ychescale9** [comment]
CI is failing because the API has changed. Can you run `./gradlew apiDump` and commit the changes?

**Comment by Komdosh** [comment]
`expireEntriesConcurrently` fails sometimes.

With 1000 repetitions almost every time it fails.

```
    @Test
    fun expireEntriesConcurrently() = runTest {
        val cache = Cache.Builder()
            .fakeTimeSource(fakeTimeSource)
            .expireAfterWrite(500.seconds)
            .build<Long, String>()

        repeat(1000) {
            cache.put(it.toLong(), "value for $it")
        }

        // should not produce a ConcurrentModificationException
        repeat(1000) {
            launch(Executors.newSingleThreadExecutor().asCoroutineDispatcher()) {
                cache.get(it.toLong())
                fakeTimeSource += 1.seconds
            }
        }
    }
```

**Comment by ychescale9** [comment]
> `expireEntriesConcurrently` fails sometimes.
> 
> With 1000 repetitions almost every time it fails.
> 
> ```
>     @Test
>     fun expireEntriesConcurrently() = runTest {
>         val cache = Cache.Builder()
>             .fakeTimeSource(fakeTimeSource)
>             .expireAfterWrite(500.seconds)
>             .build<Long, String>()
> 
>         repeat(1000) {
>             cache.put(it.toLong(), "value for $it")
>         }
> 
>         // should not produce a ConcurrentModificationException
>         repeat(1000) {
>             launch(Executors.newSingleThreadExecutor().asCoroutineDispatcher()) {
>                 cache.get(it.toLong())
>                 fakeTimeSource += 1.seconds
>             }
>         }
>     }
> ```

I'm not able to reproduce this locally and I haven't seen them failed on CI before either. It also shouldn't be a regression because you just copied it from `commonMain` to `commonTest`.

----

Let's merge this now. I'll create an issue if this starts happening more often.

**Pull request merged**